classes/Book/label.rb

```diff
@@ -1,15 +1,15 @@
 class Label
-  attr_accessor :items, :color, :title, :id
+  attr_reader :items, :id
+  attr_accessor :color, :title
 
-  def initialize(color, title, id = nil)
-    @id = id || Random.rand(1..1000)
-    @title = title
-    @color = color
+  def initialize(params)
+    @id = params[:id] || Random.rand(1..1000)
+    @title = params[:title]
+    @color = params[:color]
     @items = []
   end
 
   def add_item(item)
-    @items << item
-    item.label = self
+    @items << item unless @items.include?(item)
```

Comment:
👀 **OPTIONAL**
> Don't forget to set the `Label` instance as a property for the `Item` instance that you receive as input (the `item` local variable). As the requirements ask to implement it that way. Final version of `add_item` method could be:
```suggestion
  def add_item(item)
    @items << item unless @items.include?(item)
    item.add_label(self)
  end
```
The requirements say:
- `add_item` method in the `Label` class
  - should take an instance of the Item class as an input
    - you set that in the method definition >> `def add_item(item)`
  - should add the input item to the collection of items
    - you do this with >> `@items << item unless @items.include?(item)`
  - should add self as a property of the item object (by using the correct setter from the item object)
    - you should use the `add_label` method from `Item` instance >> `item.add_label(self)`. If `Item` doesn't have the `add_label` method we should create it. Also you should set an `attr_reader` for `:label` property at `Item` class, so the only way to set that property would be with the `add_label` method that it is called when the `Label` adds the `item` using the `add_item` method. Same thing will happen for and `Author` and a `Genre`.